File: Android/Paging-CRUD/app/src/main/java/com/hhp227/paging_crud/model/ListItem.kt
package com.hhp227.paging_crud.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

sealed class ListItem {
    @Serializable
    data class Post(
        @SerialName("id") var id: Int = 0,
        @SerialName("user_id") var userId: Int = 0,
        @SerialName("name") var name: String? = null,
        @SerialName("text") var text: String = "",
        @SerialName("status") var status: Int = 0,
        @SerialName("profile_img") var profileImage: String? = null,
        @SerialName("created_at") var timeStamp: String? = null,
        @SerialName("reply_count") var replyCount: Int = 0,
        @SerialName("like_count") var likeCount: Int = 0,
        @SerialName("report_count") var reportCount: Int = 0,
        @SerialName("attachment") var attachment: Attachment = Attachment()
    ) : ListItem()

    @Serializable
    data class Image(
        @SerialName("id") var id: Int = 0,
        @SerialName("image") var image: String? = null,
        @SerialName("tag") var tag: String? = null
    ) : ListItem()

    @Serializable
    data class Attachment(
        @SerialName("images") var imageItemList: List<Image> = emptyList(),
        @SerialName("video") var video: String? = null
    ) : ListItem()
}

File: iOS/Paging-CRUD/Paging-CRUD/Model/ListItem.swift
//
//  ListItem.swift
//  Paging-CRUD
//

import Foundation

enum ListItem {
    struct Post: Codable, Identifiable, Hashable {
        var id: Int
        var userId: Int
        var name: String?
        var text: String
        var status: Int
        var profileImage: String?
        var timeStamp: String?
        var replyCount: Int
        var likeCount: Int
        var reportCount: Int
        var attachment: Attachment

        enum CodingKeys: String, CodingKey {
            case id
            case userId = "user_id"
            case name
            case text
            case status
            case profileImage = "profile_img"
            case timeStamp = "created_at"
            case replyCount = "reply_count"
            case likeCount = "like_count"
            case reportCount = "report_count"
            case attachment
        }
    }

    struct Image: Codable, Hashable {
        var id: Int?
        var image: String?
        var tag: String?
    }

    struct Attachment: Codable, Hashable {
        var imageItemList: [Image]
        var video: String?

        enum CodingKeys: String, CodingKey {
            case imageItemList = "images"
            case video
        }
    }
}
